.github/workflows/test-windows.yml

```diff
@@ -69,22 +69,14 @@ jobs:
     - name: Print build system information
       run: python3 .github/workflows/system-info.py
 
-    - name: Install Python dependencies
-      run: >
-        python3 -m pip install
-        coverage>=7.4.2
```

Comment:
Just to call it out, there is a difference between this and https://github.com/python-pillow/Pillow/blob/82199efbf77f369ad9b7c2c3759e08f517d9925b/pyproject.toml#L57-L59

Reply:
Yep. 

* `coverage` vs `coverage>=7.4.2`: before, theoretically we could have installed coverage older than 7.4.2 here. I think that was unlikely and it's safer to require the newer version.

* `check-manifest`: we don't need that here. I see little harm in installing it here. It's in the `tests` extra because it's used by `make release-test`. If we do want to keep this env "cleaner", we could remove it from the `tests` extra and put it in something like a `release-test` extra.

Relatedly, [PEP 735](https://peps.python.org/pep-0735/) was accepted recently to define something called "dependency groups". These are like extras, but you can install a dependency group without having to also install Pillow as well.

tox and some other tools already support it, but not yet pip. We could look into using it when pip is ready.